### PR TITLE
feat(cli): exit with code 1 when remix build fails

### PR DESCRIPTION
**Pull request by gmaliar**
When builds fail remix does not exit with an error code which makes it hard to catch build failures in CI environments

Example:
```sh
❯ yarn create remix
❯ cd ./my-remix-app
❯ echo -e "import doesntExists from \"doesnt-exists\";\ndoesntExists();\n$(cat app/routes/index.tsx)" > app/routes/index.tsx
❯ npx remix build

npx remix build                                                                                                           
Building Remix app in production mode...
The path "doesnt-exists" is imported in app/routes/index.tsx but doesnt-exists is not listed in your package.json dependencies. Did you forget to install it?
 > node_modules/@web-std/file/tsconfig.json:6:24: warning: Non-relative path "packages/file/src/lib.js" is not allowed when "baseUrl" is not set (did you forget a leading "./"?)
    6 │       "@web-std/file": ["packages/file/src/lib.js"]
      ╵                         ~~~~~~~~~~~~~~~~~~~~~~~~~~


 > app/routes/index.tsx:1:25: error: Could not resolve "doesnt-exists" (mark it as external to exclude it from the bundle)
    1 │ import doesntExists from "doesnt-exists";
      ╵                          ~~~~~~~~~~~~~~~


Build failed with 1 error:
app/routes/index.tsx:1:25: error: Could not resolve "doesnt-exists" (mark it as external to exclude it from the bundle)
Built in 571ms

❯ echo $?
0
```

### CODE REVIEW

**Comment by gmaliar** [comment]
@kentcdodds can I get some feedback on this one as well? 🙏🙏🙏

File: contributors.yml [hunk 1]
```diff
@@ -159,3 +159,8 @@
 - yomeshgupta
 - zachdtaylor
 - zainfathoni
+- dima-takoy
+- anishpras
+- kumard3
+- mattmazzola
+- gmaliar
```
Comment (kentcdodds) [inline contributors.yml:166]: Should all of these people be added? Seems like you should only add your own username.

File: packages/remix-dev/cli/commands.ts [hunk 2]
```diff
@@ -60,7 +60,11 @@ export async function build(
 
   let start = Date.now();
   let config = await readConfig(remixRoot);
-  await compiler.build(config, { mode: mode, sourcemap });
+  try {
+    await compiler.build(config, { mode: mode, sourcemap });
+  } catch (error: unknown) {
```
Comment (kentcdodds) [inline packages/remix-dev/cli/commands.ts:65]: I was under the impression that a runtime error would result in a non-zero exit code.

**Review by kentcdodds — COMMENTED** [review]
Just left some thoughts. I don't merge changes to the compiler. So we really should have @mjackson and/or @jacob-ebey look at this

File: packages/remix-dev/cli/commands.ts (hunk 2 above)
Comment (gmaliar) [reply packages/remix-dev/cli/commands.ts:65]: They get swallowed here https://github.com/remix-run/remix/blob/197f570841e4813b2005d1ae007523c09df3f5c5/packages/remix-dev/compiler.ts#L307-L314

File: contributors.yml (hunk 1 above)
Comment (gmaliar) [reply contributors.yml:166]: You are correct, my bad!

**Comment by gmaliar** [comment]
@kentcdodds I accidentally deleted this branch, I made the amendments and will wait for @mjackson and/or @jacob-ebey comments here https://github.com/remix-run/remix/pull/1822

Thanks for the quick response!

**Pull request closed without merging**